src/main/java/org/mskcc/cbio/oncokb/service/MailService.java

```diff
@@ -311,18 +312,33 @@ public void sendMailToUserWhenTokenExposed(UserDTO user, ExposedToken token) {
     }
 
     @Async
-    public void sendUnapprovedUsersEmail(int daysAgo, List<UserDTO> users){
+    public void sendUnapprovedUsersEmail(int daysAgo, List<UserMessagePair> users){
         if (users == null || users.isEmpty()) {
             return;
         }
+
+        for(UserMessagePair userMessagePair : users) {
+            addUserMailsRecord(userMessagePair.getUserDTO(), LIST_OF_UNAPPROVED_USERS, applicationProperties.getEmailAddresses().getTechDevAddress(), "bot");
+        }
+
+        List<UserMessagePair> discussedUsers = users.stream().filter(user -> Objects.nonNull(user.getMessage().getReplyCount())).collect(Collectors.toList());
+        for(UserMessagePair user : discussedUsers) {
+            users.remove(user);
+            users.add(user);
+        }
```

Comment:
What's going on here?

Reply:
This sorts the list by whether or not a user has been discussed. So all of the users who have been discussed get moved to the bottom. I wasn't able to iterate through the initial list while changing the size of the list so I thought a solution would be to make a new list and iterate through that and move each corresponding element to the end.

Reply:
Should this just be a sort of the `users`? https://stackoverflow.com/a/34646172 @benzuzu

Reply:
Ah, yeah this works better. Changed and should be good now.